R/manage.R

```diff
@@ -0,0 +1,804 @@
+#' @export
+chrome_versions_remove <- function(
+  version,
+  binary,
+  platform = NULL,
+  ask = TRUE
+) {
+  force(version)
+  binary <- check_binary(binary, multiple = TRUE, allow_all = TRUE)
+  platform <- check_platform(platform, multiple = TRUE, allow_all = TRUE)
+
+  versions <- chrome_versions_list(
+    "installed",
+    binary = binary,
+    platform = platform
+  )
+
+  version <-
+    if (identical(version, "all")) {
+      versions$version
+    } else {
+      match_version(version, available_versions = versions$version)
+    }
+
+  # versions is already filtered by binary + platform
+  to_delete <- versions[versions$version %in% version, ]
+
+  dirs_delete <- chrome_versions_path_cache(
+    to_delete$version,
+    paste0(to_delete$binary, "-", to_delete$platform)
+  )
+
+  if (length(dirs_delete) == 0) {
+    cli::cli_inform("No cached binaries to remove.")
+    return(invisible())
+  }
+
+  if (!identical(ask, FALSE)) {
+    cli::cli_inform(
+      "Will remove {length(dirs_delete)} cached version{?s} of chrome:"
+    )
+    cli::cli_bullets(sprintf("{.path %s}", dirs_delete))
+
+    cli::cli_inform("Delete from cache?")
+    do_delete <- utils::menu(gettext(c("Yes", "No", "Cancel")))
+    if (do_delete != 1L) {
+      cli::cli_inform("Canceled.")
+      return(invisible(dirs_delete))
+    }
+  }
+
+  for (path_dir in dirs_delete) {
+    path_parent <- dirname(path_dir)
+    if (identical(dir(path_parent, full.names = TRUE), path_dir)) {
+      # This version contains only the binary being removed...
+      unlink(path_parent, recursive = TRUE)
+    } else {
+      unlink(path_dir, recursive = TRUE)
+    }
+  }
+
+  invisible(dirs_delete)
+}
+
+chrome_versions_ensure <- function(
+  version = "latest",
+  binary = "chrome",
+  platform = NULL,
+  prefer_installed = TRUE
+) {
+  if (identical(version, "latest-installed")) {
+    prefer_installed <- TRUE
+    version <- "latest"
+  } else if (identical(version, "latest")) {
+    prefer_installed <- FALSE
+  }
+
+  platform <- check_platform(platform)
+  binary <- check_binary(binary)
+
+  versions <- if (prefer_installed) {
+    chrome_versions_list("installed", binary = binary, platform = platform)
+  } else {
+    chrome_versions_list("all", binary = binary, platform = platform)
+  }
+
+  versions <- versions[
+    versions$binary == binary & versions$platform == platform,
+  ]
+
+  version_og <- version
+  version <- match_version(version, available_versions = versions$version)
+
+  if (is.null(version)) {
+    if (prefer_installed) {
+      return(
+        chrome_versions_ensure(
+          version_og,
+          binary = binary,
+          platform = platform,
+          prefer_installed = FALSE
+        )
+      )
+    }
+    cli::cli_abort(
+      c(
+        "Version {.field {version_og}} is not a known {.code {binary}} version.",
+        "i" = "Use {.run [chrome_versions_list()](chromote::chrome_versions_list())} to show all available versions."
+      )
+    )
```

Comment:
Cosmetic. 
Swap for early return

```suggestion
    if (!prefer_installed) {
      cli::cli_abort(
        c(
          "Version {.field {version_og}} is not a known {.code {binary}} version.",
          "i" = "Use {.run [chrome_versions_list()](chromote::chrome_versions_list())} to show all available versions."
        )
      )
    }
    return(
      chrome_versions_ensure(
        version_og,
        binary = binary,
        platform = platform,
        prefer_installed = FALSE
      )
    )
```

Reply:
Thanks for the suggestion. I like this one the way it is, and swapping the return/abort doesn't let us unnest any `if` statements.